src/vs/editor/contrib/gotoSymbol/browser/goToCommands.ts

```diff
@@ -103,7 +103,8 @@ export abstract class SymbolNavigationAction extends EditorAction2 {
 		SymbolNavigationAction._allSymbolNavigationCommands.set(opts.id, this);
 	}
 
-	override runEditorCommand(accessor: ServicesAccessor, editor: ICodeEditor, arg?: SymbolNavigationAnchor | unknown, range?: Range): Promise<void> {
+	override runEditorCommand(accessor: ServicesAccessor, editor: ICodeEditor, args: [ICodeEditor, SymbolNavigationAnchor | unknown, Range | undefined]): Promise<void> {
```

Comment:
I don't think this change is needed and I do believe that it will break existing callers

Reply:
There was a mismatch between callers and this before my PR, and it was the reason it wasn't working correctly. So either the caller in `editorExtensions.ts:455` or here needs to be changed. And the caller is more generic, so I think changing here is correct. And other inherited classes of `EditorAction2` all have an `args` argument as third argument, and no more arguments:
```
src/vs/editor/contrib/gotoSymbol/browser/goToCommands.ts
106:    override runEditorCommand(accessor: ServicesAccessor, editor: ICodeEditor, args: [ICodeEditor, SymbolNavigationAnchor | unknown, Range | undefined]): Promise<void> {

src/vs/editor/contrib/folding/browser/folding.ts
529:    public override runEditorCommand(accessor: ServicesAccessor, editor: ICodeEditor, args: T): void | Promise<void> {
src/vs/editor/browser/coreCommands.ts
36:     public runEditorCommand(accessor: ServicesAccessor | null, editor: ICodeEditor, args?: Partial<T> | null): void {
1885:           public runEditorCommand(accessor: ServicesAccessor, editor: ICodeEditor, args: unknown): void {
1961:           public runEditorCommand(accessor: ServicesAccessor, editor: ICodeEditor, args: unknown): void {
2095:           public runEditorCommand(accessor: ServicesAccessor | null, editor: ICodeEditor, args: unknown): void | Promise<void> {
2110:           public runEditorCommand(accessor: ServicesAccessor | null, editor: ICodeEditor, args: unknown): void | Promise<void> {

src/vs/editor/browser/editorExtensions.ts
276:                    public runEditorCommand(accessor: ServicesAccessor, editor: ICodeEditor, args: any): void {
377:    public runEditorCommand(accessor: ServicesAccessor, editor: ICodeEditor, args: any): void | Promise<void> {

src/vs/editor/contrib/wordOperations/browser/wordOperations.ts
44:     public runEditorCommand(accessor: ServicesAccessor, editor: ICodeEditor, args: any): void {
333:    public runEditorCommand(accessor: ServicesAccessor, editor: ICodeEditor, args: any): void {

src/vs/editor/contrib/codeAction/browser/codeActionCommands.ts
210:    public runEditorCommand(_accessor: ServicesAccessor, editor: ICodeEditor, userArgs: any) {

src/vs/workbench/contrib/testing/browser/testingOutputPeek.ts
1675:   public runEditorCommand(accessor: ServicesAccessor, editor: ICodeEditor) {
1705:   public runEditorCommand(accessor: ServicesAccessor, editor: ICodeEditor) {
1723:   public runEditorCommand(accessor: ServicesAccessor, editor: ICodeEditor) {
1750:   public runEditorCommand(accessor: ServicesAccessor, editor: ICodeEditor) {
```

Reply:
You are right. Tho, I would fix this at the few special callers that we have. Usually these commands are invoked via the "normal command execution" but some are special: namely inlay hint and definition links. [This](https://github.com/microsoft/vscode/blob/356b57c157a39ef957bca466240711e682ccca54/src/vs/editor/contrib/gotoSymbol/browser/link/goToDefinitionAtPosition.ts#L304) and [this](https://github.com/microsoft/vscode/blob/ab1a31add4608f484d555e3c791f03321a4dda5f/src/vs/editor/contrib/inlayHints/browser/inlayHintsLocations.ts#L108) invocation should be changed so that the editor isn't passed anymore